### PR TITLE
Manual Settlement for Link Chat

### PR DESCRIPTION
**Pull request by antidmg**
This adds the ability to start and handle settlements with direct neighbors via link chat.

In the chat UI loop, you can now use the commands `!settle <amount in microMEL>`, and the other side can respond with `!accept` or `!reject`.

Currently, the only type of "proof" you can provide is manual and trust-based.

### CODE REVIEW

File: src/daemon/inout_route/chat.rs [hunk 1]
```diff
@@ -92,15 +97,99 @@ pub fn get_chat(ctx: &DaemonContext, neigh: Fingerprint) -> Vec<(bool, String, S
...
+    ctx: &DaemonContext,
+    dest: Fingerprint,
+    msg: String,
+) -> anyhow::Result<()> {
     let chats = ctx.get(CHATS);
+    let my_sk = *ctx.get(GLOBAL_IDENTITY);
+    let settlements = ctx.get(SETTLEMENTS);
 
-    if let Some(client) = chats.clients.get(&dest) {
+    if msg.starts_with("!settle ") {
```
Comment (antidmg) [inline src/daemon/inout_route/chat.rs:109]: Maybe we can use an enum with a custom impl for `FromStr` instead of hardcoding the strings?

File: src/daemon/inout_route/chat.rs [hunk 2]
```diff
@@ -92,15 +97,99 @@ pub fn get_chat(ctx: &DaemonContext, neigh: Fingerprint) -> Vec<(bool, String, S
...
+                            .bold()
+                    }
+                    Err(e) => format!("error sending <{dest}> a settlement request: {e}")
+                        .red()
+                        .bold(),
+                };
+
+                chats.insert(
+                    dest,
+                    ChatEntry::new_incoming(format!("{}", res_msg.bold().blue())),
```
Comment (nullchinchilla) [inline src/daemon/inout_route/chat.rs:169]: After a bit more thought, I think a better design for the chat protocol overall would be to not have the logic here, but rather in the control protocol *client*, and expose separate verbs in the control protocol for manual settlement.

The reason is that the localhost control protocol should be useful for a variety of frontends --- for instance, a GUI frontend that has separate panels for chatting and settling debts.

Thus, stuff like coloring responses, etc, should be done on the frontend. The control protocol server should return UI-agnostic JSON representations and such, not strings to be directly displayed on the GUI.

For now, we can merge it in, since changing this whole approach is something that will touch on other parts of the control protocol too.

File: src/daemon/inout_route/chat.rs [hunk 3]
```diff
@@ -35,7 +40,7 @@ static CHATS: CtxField<Chats> = |ctx| {
     })
 };
 
-pub fn incoming_chat(ctx: &DaemonContext, neighbor: Fingerprint, msg: String) {
+pub async fn incoming_chat(ctx: &DaemonContext, neighbor: Fingerprint, msg: String) {
```
Comment (nullchinchilla) [inline src/daemon/inout_route/chat.rs:43]: Hmm why was this changed to async? There's no `.await` in it, so this is a bit misleading (since it indicates to the caller that this function could possibly "block")

File: src/daemon/settlement.rs [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+use smol::channel::{Receiver, Sender};
+use stdcode::StdcodeSerializeExt;
+
+use super::context::{DaemonContext, DEBTS, GLOBAL_IDENTITY};
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct SettlementRequest {
+    // unix milliseconds
+    timestamp_ms: u64,
+    pub decrease: u64,
```
Comment (nullchinchilla) [inline src/daemon/settlement.rs:22]: this does not need to be `pub`? The only place that this field is used is in the same module.

Note that public/private visibility in Rust is at the *module* level, not at the type level, unlike OOP languages. So it's perfectly fine for a method of Settlements to call something private in SettlementRequest as long as they're in the same module.

**Review by nullchinchilla — CHANGES_REQUESTED** [review]
We can fix the larger structural issue later; other than that it's minor things :)
